File: examples/tags/GeneSEZ_Examples_2-1-0/org.genesez.example.csharp.forms/src-gen/forms/Point.cs
namespace forms
{
	#region Imports
	using System;
	/* PROTECTED REGION ID(java.mclassifier.other.import.code_14_0_b6f02e1_1211871095796_723781_254) ENABLED START */
	/* TODO put your imports here */

	/* PROTECTED REGION END */
	#endregion
	
	public class Point : IForm
	{
		#region Private Fields
		private    double x;
		private    double y;
		#endregion
		
		#region Properties
		public  double X
		{

			get { return this.x; }
			set { this.x = value; }
		}
		public  double Y
		{

			get { return this.y; }
			set { this.y = value; }
		}
		#endregion
		
		#region Constructors
	public Point( double x,  double y)
		{
			/* PROTECTED REGION ID(java.moperation.implementation.operation.code._14_0_b6f02e1_1211871240109_473577_277) ENABLED START */
			/* TODO put your own implementation code here */
			this.x = x;
			this.y = y;
			/* PROTECTED REGION END */
		}
		
		#endregion
		
		#region Your Own Implementations
		/* PROTECTED REGION ID(java.mclassifier.implementation.owncode.code._14_0_b6f02e1_1211871095796_723781_254) ENABLED START */
		/* TODO put your own implementation code here */

		public void Move( double dx, double dy )
		{
			this.x += dx;
			this.y += dy;
		}
        		
        public void Rotate(Point c, double phi)
		{
			double phiAsRadian = (phi * Math.PI) / 180;
			double tx = this.x - c.x;
			double ty = this.y - c.y;
			double h = Math.Sqrt( Math.Pow( tx, 2 ) + Math.Pow( ty, 2 ));
			double dx = Math.Cos( phiAsRadian ) * h;
			double dy = Math.Sin( phiAsRadian ) * h;
			this.x = dx + c.x;
			this.y = dy + c.y;
		}

		/* PROTECTED REGION END */
		#endregion
	}

}

File: examples/branches/GeneSEZ_Examples_2-1-0/org.genesez.example.java.forms/src-gen/forms/Point.java
package forms;

/* 
 *	Do not place import/include statements above this comment, just below. 
 * 	@FILE-ID : (_14_0_b6f02e1_1211871095796_723781_254) 
 */

import java.awt.Color;

/**
 * Describes a <i><b>point</b></i> in a 2-dimensional coordinate system
 * @author dreamer
 */
public class Point implements Form {
	
	// -- generated attribute, constant + association declarations ----------
	
	/**
	 * the x coordinate value
	 */
	private double x;
	
	/**
	 * the y coordinate value
	 */
	private double y;
	
	// -- generated constructors --------------------------------------------
	/**
	 * constructs a point in the center of the coordinate system
	 */
	public Point() {
		/* PROTECTED REGION ID(java.constructor._14_0_b6f02e1_1211871240109_473577_277) ENABLED START */
		this.x = 0.0;
		this.y = 0.0;
		/* PROTECTED REGION END */
	}
	
	/**
	 * constructs a point using the specified coordinate values
	 * @param	x	the x-coordinate value of the point
	 * @param	y	the y-coordinate value of the point
	 */
	public Point(double x, double y) {
		/* PROTECTED REGION ID(java.constructor._16_0_b6f02e1_1249463853031_881157_741) ENABLED START */
		this.x = x;
		this.y = y;
		/* PROTECTED REGION END */
	}
	
	// -- generated method stubs for implementations + derived attributes ---
	/**
	 * determines the center of area of the form
	 * @return	the location of the center of area
	 */
	public Point getCenterOfArea() {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__16_0_b6f02e1_1249464477484_822826_834) ENABLED START */
		return new Point(this.x, this.y);
		/* PROTECTED REGION END */
	}
	
	/**
	 * determines if the form is crossing another form
	 * @param	form	the form to check for crossing
	 * @return	true, if the specified form crosses this form, otherwise false
	 */
	public boolean isCrossing(Form form) {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__16_0_b6f02e1_1249464485156_784670_838) ENABLED START */
		// TODO: implementation of method 'Point.isCrossing(...)'
		throw new UnsupportedOperationException("The implementation of this generated method stub is missing!");
		/* PROTECTED REGION END */
	}
	
	/**
	 * moves the position by the given delta values within a 2-dimensional 
	 * coordinate system
	 * @param	dx	delta value to move in <i><b>x</b></i> direction
	 * @param	dy	delta value to move in <i><b>y</b></i> direction
	 */
	public void move(double dx, double dy) {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__14_0_b6f02e1_1214478865734_965960_209) ENABLED START */
		// TODO: implementation of method 'Point.move(...)'
		throw new UnsupportedOperationException("The implementation of this generated method stub is missing!");
		/* PROTECTED REGION END */
	}
	
	/**
	 * performs a rotation by the specified angle
	 * @param	phi	the angle in degrees
	 */
	public void rotate(double phi) {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__14_0_b6f02e1_1212580097609_734698_127) ENABLED START */
		// TODO: implementation of method 'Point.rotate(...)'
		throw new UnsupportedOperationException("The implementation of this generated method stub is missing!");
		/* PROTECTED REGION END */
	}
	
	/**
	 * performs a rotations around the specified point by the specified angle
	 * @param	center	the center of the rotation
	 * @param	phi	the angle in degrees
	 */
	public void rotate(Point center, double phi) {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__14_0_b6f02e1_1214478889937_90346_236) ENABLED START */
		// TODO: implementation of method 'Point.rotate(...)'
		throw new UnsupportedOperationException("The implementation of this generated method stub is missing!");
		/* PROTECTED REGION END */
	}
	
	/**
	 * changes the color by repainting using the specified color
	 * @param	color	the color to use for repainting
	 */
	public void repaint(Color color) {
		/* PROTECTED REGION ID(java.implementation._14_0_b6f02e1_1211871095796_723781_254__16_0_b6f02e1_1249464738828_388422_1023) ENABLED START */
		// TODO: implementation of method 'Point.repaint(...)'
		throw new UnsupportedOperationException("The implementation of this generated method stub is missing!");
		/* PROTECTED REGION END */
	}
	
	// -- generated association + attribute accessors -----------------------
	/**
	 * the x coordinate value
	 */
	public double getX() {
		return x;
	}
	
	/**
	 * the y coordinate value
	 */
	public double getY() {
		return y;
	}
	
	// -- generated code of other cartridges --------------------------------
	
	// -- own code implementation -------------------------------------------
	/* PROTECTED REGION ID(java.class.own.code.implementation._14_0_b6f02e1_1211871095796_723781_254) ENABLED START */
	// TODO: put your own implementation code here
	/* PROTECTED REGION END */
	
}
